app/scripts/shell/dimAppCtrl.controller.js

```diff
@@ -124,21 +124,41 @@
      * Show a popup dialog containing the given template. Its class
      * will be based on the name.
      */
-    function showPopupFunction(name) {
+    function showPopupFunction(name, lang) {
       var result;
       return function(e) {
         e.stopPropagation();
 
+        if (lang) {
+          lang = vm.settings.language + '/'; // set language
+        }
+
         if (result) {
           result.close();
         } else {
           ngDialog.closeAll();
-          result = ngDialog.open({
-            template: 'views/' + name + '.html',
-            className: name,
-            appendClassName: 'modal-dialog'
-          });
-
+          if (lang) {
+            result = ngDialog.open({
+              template: 'views/' + lang + name + '.html',
+              className: name,
+              name: name,
+              appendClassName: 'modal-dialog'
+            });
+            /* Lang Fallback
+            result = ngDialog.open({
+              template: 'views/en/' + name + '.html',
+              className: name,
+              name: name,
+              appendClassName: 'modal-dialog'
+            }); */
+          } else {
+            result = ngDialog.open({
+              template: 'views/' + name + '.html',
+              className: name,
+              name: name,
+              appendClassName: 'modal-dialog'
+            });
```

Comment:
Rather than copy/pasting two blocks of code, just set `lang` to `''` if it's unset.